### PR TITLE
fix: extract give_receipt shortcode from post content

### LINKED ISSUE #5042: Give Receipt shortcodes should work in page builders
## User Story

As an Admin, I would like to be able to use my `[give_receipt]` in a page builder/editor, so that my users can still access their donation receipts if I'm not pasting the shortcode into the standard WP editor

## Details

Donation receipt doesn't render well when using page builders like Divi. 

## Steps to Reproduce

1. Install and activate Divi  page builder plugin
2. Go to the Donation Success page and edit page with Divi editor
3. Insert `[give_receipt]` shortcode into Divi Code Module  and hit save
4. View page on the front-end

## Visuals

image

## Acceptance Criteria

-   [x]  `[give_receipt]`  shortcodes work in page builders
-   [x]  Receipt is rendered on the front-end without issues

### PR DESCRIPTION
**Pull request by alaca**
Resolves #5042

## Description

The current [method](https://github.com/impress-org/givewp/blob/3bb91d25819daf834d6f5212de48e9a1fe854810/src/Helpers/Frontend/Shortcode.php#L12) for extracting **give_receipt** shortcode from the post content, sometimes cannot extract the shortcode if the multiple shortcodes are being used ( i.e when using Divi page builder ). This PR solves this issue by using a regex pattern focused on **give_receipt** string which will be more efficient and possibly less prone to errors.
## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

## Testing Instructions

   1. Install and activate Divi page builder plugin
   2. Go to the Donation Success page and edit page with Divi editor
   3. Insert `[give_receipt]` shortcode into Divi Code Module and hit save
   4. View page on the front-end

### CODE REVIEW

**Comment by kevinwhoffman** [comment]
@alaca I see that you mentioned Divi in the testing instructions. Were you able to confirm this fix in any other page builder?

**Comment by alaca** [comment]
@kevinwhoffman yes, WP Bakery.

**Comment by alaca** [comment]
@DevinWalker it took me some time to figure out what is happening here. The "problem" is you have inserted [give_receipt] shortcode 2 times into the Donation Confirmation page. That's why you have two Donation Receipts ( Your video at 3:57 ). When using the Legacy form to make a donation, and when the form redirects you to the Donation Confirmation screen, only the first Donation Receipt is shown, the second one is not even loaded into the page. That's why you see that animation/loader below the first Donation Receipt. Why this is happening in the first place is another issue.

**Comment by DevinWalker** [comment]
@alaca sorry to send you down a rabbit hole there! I just tested again and this is working fine.

File: src/Helpers/Frontend/Shortcode.php [hunk 1]
```diff
@@ -10,17 +10,15 @@ class Shortcode {
...
 
-		if (
-			! empty( $content ) &&
-			preg_match_all( '/' . $pattern . '/s', $content, $matches ) &&
-			array_key_exists( 2, $matches ) &&
-			in_array( 'give_receipt', $matches[2], true )
-		) {
-			return $matches[0][0];
+		if ( ! empty( $content ) ) {
+			preg_match( '/\[give_receipt(.*?)]/', $content, $matches );
```
Comment (JasonTheAdams) [inline src/Helpers/Frontend/Shortcode.php:17]: Let's add the insensitive flag here to make sure that's not an issue.
```suggestion
			preg_match( '/\[give_receipt(.*?)]/i', $content, $matches );
```

**Comment by DevinWalker** [comment]
Tested again and it works fine.

**Pull request merged**